source/Sgghost.h

```diff
@@ -0,0 +1,139 @@
+  double _intval =0.0, emp::vector<emp::Ptr<Organism>> _syms = {},
+  emp::vector<emp::Ptr<Organism>> _repro_syms = {},
+  std::set<int> _set = std::set<int>(),
+  double _points = 0.0) : Host(_random, _world, _config, _intval,_syms, _repro_syms, _set, _points) { 
+    if ( _intval > 1 || _intval < -1) {
+       throw "Invalid interaction value. Must be between -1 and 1";  // Exception for invalid interaction value
+     };
+   }
+
+
+
+  sggHost(const sggHost &) = default;
+  sggHost(sggHost &&) = default;
+  sggHost() = default;
+
+  sggHost & operator=(const sggHost &) = default;
+  sggHost & operator=(sggHost &&) = default;
+  bool operator==(const sggHost &other) const { return (this == &other);}
+  bool operator!=(const sggHost &other) const {return !(*this == other);}
+
+
+ 
+  double GetPool() {return sourcepool;}
+  void SetPool(double _in) {sourcepool= _in;}
+  void AddPool(double _in) {sourcepool += _in;}
+
+
+
+
+
+
+  void DistribResources(double resources) {
+    double hostIntVal = interaction_val; //using private variable because we can
+
+    //In the event that the host has no symbionts, the host gets all resources not allocated to defense or given to absent partner.
+    if(syms.empty()) {
+
+      if(hostIntVal >= 0){
+	      double spent = resources * hostIntVal;
+        this->AddPoints(resources - spent);
+      }
+      else {
+        double hostDefense = -1.0 * hostIntVal * resources;
+        this->AddPoints(resources - hostDefense);
+      }
+      return; //This concludes resource distribution for a host without symbionts
+    }
+
+    //Otherwise, split resources into equal chunks for each symbiont
+    int num_sym = syms.size();
+    double sym_piece = (double) resources / num_sym;
+   
+
+    for(size_t i=0; i < syms.size(); i++){
+      double hostPortion = syms[i]->ProcessResources(sym_piece);
+      double hostPool = syms[i]->ProcessPool();
+      this->AddPoints(hostPortion);
+      this->AddPool(hostPool);
+    }
+    if(syms.size()>0){this->DistribPool();}
+    
+
+  } //end DistribResources
+
+  void DistribPool(){
+    //to do: marginal return
+    int num_sym = syms.size();
+    double sym_piece = (double) sourcepool / num_sym;
+    for(size_t i=0; i < syms.size(); i++){
+        syms[i]->AddPoints(sym_piece);
+    }
+    this->SetPool(0);
+  }
+
+  void Process(size_t location) {
+    //Currently just wrapping to use the existing function
+    double resources = my_world->PullResources();
+    DistribResources(resources);
+    // Check reproduction
+    if (GetPoints() >= my_config->HOST_REPRO_RES() && repro_syms.size() == 0) {  // if host has more points than required for repro
+        // will replicate & mutate a random offset from parent values
+        // while resetting resource points for host and symbiont to zero
+        emp::Ptr<sggHost> host_baby = emp::NewPtr<sggHost>(random, my_world, my_config, GetIntVal());
```

Comment:
Is the type of `host_baby` the only reason we can't use the version of `Process` implemented in `Host`? That seems unfortunate, but all of the solutions are probably more trouble than they're worth.